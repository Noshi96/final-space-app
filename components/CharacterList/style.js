import styled from 'styled-components'

export const List = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 2rem;

  @media only screen and (min-width: 48rem) {
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: center;
  }
`
