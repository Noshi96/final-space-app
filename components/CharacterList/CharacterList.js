import { List } from './style'
import CharacterCard from '../CharacterCard/CharacterCard'

const CharacterList = ({ characters }) => {
  if (!characters || characters.length === 0) return null

  return (
    <List>
      {characters.map((singleCharacter) => {
        return (
          <CharacterCard
            key={singleCharacter.id}
            name={singleCharacter.name}
            imgUrl={singleCharacter.img_url}
          />
        )
      })}
    </List>
  )
}

export default CharacterList
