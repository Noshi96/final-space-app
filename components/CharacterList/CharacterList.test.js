/* globals describe, it, expect  */

import { render } from '@testing-library/react'
import CharacterList from './CharacterList'

describe('CharacterList', () => {
  const characters = [
    {
      id: '1',
      name: 'Gary Goodspeed',
      imgUrl:
        'https://finalspaceapi.com/api/character/avatar/gary_goodspeed.png',
    },
  ]

  it('renders character list', () => {
    const characterList = render(<CharacterList characters={characters} />)
    expect(characterList.container).toMatchSnapshot()
  })
})
