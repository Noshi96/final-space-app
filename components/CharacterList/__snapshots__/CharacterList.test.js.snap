// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterList renders character list 1`] = `
<div>
  <div
    class="sc-e4f7e63c-0 gaCDvv"
  >
    <div
      class="sc-7786e022-0 hGLfzY"
    >
      <img
        alt="Gary Goodspeed"
        class="sc-7786e022-2 cLquHY"
      />
      <div
        class="sc-7786e022-1 dtZBjv"
      >
        Gary Goodspeed
      </div>
    </div>
  </div>
</div>
`;
